### DISCUSSION TITLE
Baseline Container Image Version

### DISCUSSION THREAD

**Discussion by RyanPotts** [prometheus-operator/prometheus-operator#3891] [Q&A]
Hello, how can I find what version of Prometheus is being used for the Operator? The current version of Prometheus is 2.25 but the current version of the Prometheus operator is 0.45.

I understand Operators have their own versions based on manifests and metadata.

**Comment by simonpasquier** [comment]
All supported versions are listed here: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/compatibility.md#prometheus

**Reply by RyanPotts** [reply]
Thank you

Is there a way to set a compatible Prometheus version before running the Operator?

**Reply by simonpasquier** [reply]
not sure what you mean, can you clarify?

**Reply by RyanPotts** [reply]
We can use Prometheus as an example: When an operator is created it is essentially extra mainfests/CRD's and some pieces of software (correct me if I'm wrong). The Prometheus Operator will need to use a Prometheus container image (or tar file, etc.). Short of installing the Prometheus Operator and running an "oc debug" against the Prometheus Pod to tell what version of Prometheus the Operator is using - is there another way to tell (or set) what version of Prometheus will be used for the Operator?

**Reply by paulfantom** [reply]
Yes, you specify which Prometheus image you want to use and which version in Prometheus CR. For example:
- image - https://github.com/thaum-xyz/ankhmorpork/blob/master/apps/monitoring/manifests/prometheus/prometheus.yaml#L37
- version - https://github.com/thaum-xyz/ankhmorpork/blob/master/apps/monitoring/manifests/prometheus/prometheus.yaml#L75

You can get a detailed explanation of each Prometheus CR field at https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#prometheusspec

Additionally prometheus and prometheus-operator emit `*_build_info` metrics which point what is the currently running version of the component.
